lambdas/data-migration2/tests/test-pdrs.js

```diff
@@ -0,0 +1,344 @@
+  await t.throwsAsync(
+    migratePdrRecord(testPdr, knex),
+    { name: 'SchemaValidationError' }
+  );
+});
+
+test.serial('migratePdrRecord handles nullable fields on source PDR data', async (t) => {
+  const {
+    collectionCumulusId,
+    knex,
+    pdrPgModel,
+    providerCumulusId,
+    testCollection,
+    testProvider,
+  } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.name);
+
+  delete testPdr.execution;
+  delete testPdr.PANSent;
+  delete testPdr.PANmessage;
+  delete testPdr.progress;
+  delete testPdr.stats;
+  delete testPdr.address;
+  delete testPdr.originalUrl;
+  delete testPdr.timestamp;
+  delete testPdr.duration;
+  delete testPdr.updatedAt;
+
+  await migratePdrRecord(testPdr, knex);
+  const record = await pdrPgModel.get(knex, { name: testPdr.pdrName });
+
+  t.like(
+    record,
+    {
+      name: testPdr.pdrName,
+      provider_cumulus_id: providerCumulusId,
+      collection_cumulus_id: collectionCumulusId,
+      execution_cumulus_id: null,
+      status: testPdr.status,
+      progress: null,
+      pan_sent: null,
+      pan_message: null,
+      stats: null,
+      address: null,
+      original_url: null,
+      timestamp: null,
+      duration: null,
+      created_at: new Date(testPdr.createdAt),
+    }
+  );
+});
+
+test.serial('migratePdrRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+  const { knex, testCollection, testProvider } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.name);
+  await migratePdrRecord(testPdr, knex);
+
+  await t.throwsAsync(
+    migratePdrRecord(testPdr, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migratePdrs skips already migrated record', async (t) => {
+  const {
+    knex,
+    testCollection,
+    testProvider,
+  } = t.context;
+  const testPdr = generateTestPdr(testCollection.name, testProvider.name);
+
+  await migratePdrRecord(testPdr, knex);
+  await pdrsModel.create(testPdr);
+
+  const migrationSummary = await migratePdrs(process.env, knex);
+  t.deepEqual(migrationSummary,
+    {
+      dynamoRecords: 1,
+      skipped: 1,
+      failed: 0,
+      success: 0,
+    });
+
+  const records = await knex(tableNames.pdrs).where({ name: testPdr.pdrName });
+  t.is(records.length, 1);
+  t.teardown(() => pdrsModel.delete({ pdrName: testPdr.pdrName }));
+});
+
+test.serial('migratePdrs processes multiple PDR records', async (t) => {
+  const {
+    collectionPgModel,
+    knex,
+    providerPgModel,
+    testCollection,
+    testProvider,
+  } = t.context;
+
+  const fakeCollection = fakeCollectionRecordFactory();
+  const fakeProvider = fakeProviderRecordFactory();
+  await collectionPgModel.create(knex, fakeCollection);
+  await providerPgModel.create(knex, fakeProvider);
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.name);
+  const anotherPdr = generateTestPdr(fakeCollection.name, fakeProvider.name);
```

Comment:
I could be wrong, but do you really need to create a separate collection and provider for this second PDR? I think you could just re-use the `testCollection` and `testProvider` to reduce the test setup here

Reply:
Oh, for some reason I thought that the collection and provider that a PDR references had to be unique between PDRs but that doesn't seem to be the case.